### ISSUE TITLE
API route leak content of errors

### ISSUE THREAD

**Issue by cram0** [nuxt/nuxt#14491]
### Environment

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.15.0`
- Nuxt Version:     `3.0.0-rc.6`
- Package Manager:  `npm@8.5.5`
- Builder:          `vite`
- User Config:      `vite`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

1. Clone this repo :

[https://github.com/cram0/poc_secret](https://github.com/cram0/poc_secret)

2. Install dependencies :

`npm i`

3. Run build and preview

`npm run build && npm run preview`

4. Once the server is running, do a POST request with cURL :

`curl -X POST -H 'Content-Type: application/json' -d '{}' http://127.0.0.1:3000/api/test`

You should see the message returned by the server which is the thrown error : 

`{"url":"/api/test","statusCode":500,"statusMessage":"Internal Server Error","message":"Hello secret error","description":""}`

And the error thrown by the server : 
`[nuxt] [request error] Hello secret error
at createError (./server/node_modules/h3/dist/index.mjs:191:12)
at Server.nodeHandler (./server/node_modules/h3/dist/index.mjs:386:21)`

### Describe the bug

Basically the Nuxt server returns the error thrown inside of API routes using h3

### Logs

```shell
[h3] [unhandled] H3Error: Hello secret error
    at createError (file:///xxx/poc_secret/.output/server/node_modules/h3/dist/index.mjs:191:12)
    at Server.nodeHandler (file:///xxx/poc_secret/.output/server/node_modules/h3/dist/index.mjs:386:21) {
  statusCode: 500,
  fatal: false,
  unhandled: true,
  statusMessage: 'Internal Server Error'
}
[nuxt] [request error] Hello secret error
  at createError (./server/node_modules/h3/dist/index.mjs:191:12)  
  at Server.nodeHandler (./server/node_modules/h3/dist/index.mjs:386:21)
```

**Comment by OrbisK** [comment]
The responses of unhandled errors are looking fine with latest nuxt:

```
{
  "error": true,
  "url": "http://127.0.0.1:3000/api/test",
  "statusCode": 500,
  "statusMessage": "Server Error",
  "message": "Server Error"
}
```
